### PR TITLE
Rerender for looser libxcb pin

### PR DESCRIPTION
**Pull request by h-vetinari**
I don't know how https://github.com/conda-forge/cairo-feedstock/commit/99df405140fcc99230b9540a2def34f19842fa57 was produced, but it missed a ~3 month old [change](https://github.com/conda-forge/conda-forge-pinning-feedstock/commit/6ad452344709c015409a8a7f5f442e97f774ea25) from the global pinning that's crucial to being able to install cairo next to current `xorg-*` packages (e.g. https://github.com/conda-forge/openjdk-feedstock/pull/175)

```
Could not solve for environment specs
The following packages are incompatible
├─ harfbuzz =10 * is installable and it requires
│  └─ cairo >=1.18.2,<2.0a0 *, which requires
│     └─ libxcb >=1.16,<1.17.0a0 *, which can be installed;
└─ xorg-libxext >1.3.4 * is not installable because it requires
   └─ xorg-libx11 >=1.8.10,<2.0a0 *, which requires
      └─ libxcb >=1.17.0,<2.0a0 *, which conflicts with any installable versions previously reported.
```

### CODE REVIEW

**Comment by h-vetinari** [comment]
Seems that something about our "new" xorg stack breaks the detection of `xlib` here
```
    Xlib                    : NO
    Xlib Xrender            : NO
    Quartz                  : NO
    Quartz-image            : NO
    XCB                     : YES
```
leading to missing `cairo-xlib.pc`. That's despite `xorg-libx11` in host, which does [have](https://conda-metadata-app.streamlit.app/?q=conda-forge%2Flinux-64%2Fxorg-libx11-1.8.10-h4f16b4b_1.conda&with_broken=true) `include/X11/Xlib.h` (for example).

Any ideas what's happening here @hmaarrfk @pkgw?

**Comment by hmaarrfk** [comment]
thanks!

**Comment by hmaarrfk** [comment]
One now needs `xorg-xorgproto` for the headers. The new linting hint should have caught that, unless it doesn't catch "build" requirements.

File: recipe/meta.yaml
Comment (hmaarrfk) [inline recipe/meta.yaml:31]: These xorg requirements in the build section are dubious, are they truely necessary?

File: recipe/meta.yaml
Comment (hmaarrfk) [inline recipe/meta.yaml:47]: The changes actually remove `zlib` from the runtime requirements. Its somewhat annoying to start to pull in our whole "build time requirements" packages at runtime for space and simplicity considerations.

We should likely move to a replacement for `pkg-config` instead that don't require privately linked library at compile time to "find" the package.

**Review by hmaarrfk — APPROVED** [review]

**Comment by h-vetinari** [comment]
> The new linting hint should have caught that, unless it doesn't catch "build" requirements.

It didn't (and I had double-checked that), because the build never explicitly depended on the packages that got the lint. I guess they got pulled in transitively

File: recipe/meta.yaml
Comment (h-vetinari) [reply recipe/meta.yaml:31]: I can't really speak to the needs of this recipe or the details of the new xorg setup...

Comment (hmaarrfk) [reply recipe/meta.yaml:31]: yeah once the recipe is building, lets try to slim these down before we merge on main

**Comment by hmaarrfk** [comment]
ok fingers crossed. otherwise we should keep those libraries in the build section.

**Comment by hmaarrfk** [comment]
linux_64:
```
cairo 1.18.2

  Surface Backends
    Image                   : YES
    SVG                     : YES

  Font Backends
    User                    : YES
    FreeType                : YES
    Fontconfig              : YES
    Win32                   : NO
    Win32 DWrite            : NO
    Quartz                  : NO

  Functions
    PNG functions           : YES
    X11-xcb                 : YES
    XCB-shm                 : YES

  Features and Utilities
    cairo-trace:            : YES
    cairo-script-interpreter: YES
    API reference           : NO

  User defined options
    buildtype               : release
    default_library         : both
    fontconfig              : enabled
    freetype                : enabled
    glib                    : enabled
    libdir                  : lib
    prefix                  : $PREFIX
    wrap_mode               : nofallback
    xlib-xcb                : enabled

```


linux_aarch64
```
cairo 1.18.2

  Surface Backends
    Image                   : YES
    PDF                     : YES
    SVG                     : YES

  Font Backends
    User                    : YES
    FreeType                : YES
    Fontconfig              : YES
    Win32                   : NO
    Win32 DWrite            : NO
    Quartz                  : NO

  Functions
    PNG functions           : YES
    X11-xcb                 : YES
    XCB-shm                 : YES

  Features and Utilities
    cairo-trace:            : YES
    cairo-script-interpreter: YES
    API reference           : NO

  User defined options
    Cross files             : $BUILD_PREFIX/meson_cross_file.txt
                              cross_file.txt
    buildtype               : release
    default_library         : both
    fontconfig              : enabled
    freetype                : enabled
    glib                    : enabled
    libdir                  : lib
    prefix                  : $PREFIX
    wrap_mode               : nofallback
    xlib-xcb                : enabled

```

Looks right to me.

File: recipe/meta.yaml
Comment (hmaarrfk) [inline recipe/meta.yaml:31]:
```suggestion
```

If the logs look good to you @h-vetinari we can probably push this last cleanup and add automerge.

**Review by h-vetinari — COMMENTED** [review]
Can't approve my own PR, but this LGTM! Thanks for the quick help!

**Review by hmaarrfk — APPROVED** [review]
I'm not a maintainer here but I've been working with many of the maintainers to get these xorg recipes improved. I also feel like this libxcb got missed by a migrator that loosened the pin alltogether. so it is pretty safe to merge IMO.

**Pull request merged**
